src/pages/admin/Event/EventEdit/EventEditCheckBoxFields.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+
+import {
+  Button,
+  Grid,
+  Checkbox,
+  Typography,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import DeleteIcon from '@material-ui/icons/Delete';
+import CustomTextField from "components/inputs/CustomTextField";
+
+const useStyles = makeStyles(() => ({
+  table: {
+    width: "550px"
+  },
+  deleteIcon: {
+    cursor: "pointer"
+  }
+}));
+
+export default function EventEditCheckBoxFields(props) {
+  const classes = useStyles();
+  const [isRequired, setIsRequired] = useState(false);
+  const [currCheckBoxField, setCurrCheckBoxField] = useState("");
+
+  const {
+    initialValues,
+    setInitialValues
+  } = props;
+
+  const handleCheckBoxField = (e) => {
+    e.persist();
+    setCurrCheckBoxField(e.target.value);
+  };
+
+  const handleAddCheckBoxField = () => {
+    initialValues.checkBoxFields.push({
+      value: currCheckBoxField,
+      isRequired
+    });
+    setInitialValues({
+      ...props.initialValues,
+    })
+    setCurrCheckBoxField("");
+    setIsRequired(false);
+  };
+
+  return (
+    <React.Fragment>
+      <Typography>Note: required checkboxes are boxes that MUST be checked for the form to be submitted</Typography>
+      <br />
+      <br />
+      <Grid container spacing={3}>
+        <Grid item xs={6}>
+          <CustomTextField
+            {...props}
+            label="Add a CheckBox Field"
+            groupName="currCheckBoxField"
+            value={currCheckBoxField}
+            handleEvent={(event) => handleCheckBoxField(event)} />
+        </Grid>
+
+        <Grid item xs={6}>
+            <Typography>Field required?</Typography>
+            <Checkbox
+              checked={isRequired}
+              onChange={() => setIsRequired(!isRequired)}
+              color="primary" />
+        </Grid>
+        <Grid item xs={6}>
+        </Grid>
+      </Grid>
+      <Button variant="contained" onClick={() => {handleAddCheckBoxField()}}>Add Field</Button>
+
+      <Table className={classes.table} overflow="hidden">
+          <TableHead>
+              <TableRow>
+                  <TableCell style={{color: "white"}}>Select Field</TableCell>
+                  <TableCell style={{color: "white"}} align="right">Is Required</TableCell>
+                  <TableCell />
+              </TableRow>
+          </TableHead>
+
+          <TableBody>
+              {initialValues.checkBoxFields.map((checkBoxField, index) => (
```

Comment:
Should we wrap this in a useMemo, so it doesn't get rerendered unnecessarily

Reply:
![image](https://user-images.githubusercontent.com/45786074/121804357-cb6b4700-cbfa-11eb-8d24-d96e398e6631.png)

think it might be ok to leave it since it's the exec side of the app so it shouldn't get tOO crazy, thoughts?

Reply:
Goood point sir